pkg/kf/testutil/integration.go

```diff
@@ -52,25 +62,27 @@ func DockerRegistry() string {
 	return os.Getenv(EnvDockerRegistry)
 }
 
-// RunIntegrationTest skips the tests if testing.Short() is true (via --short
-// flag) or if DOCKER_REGISTRY is not set. Otherwise it runs the given test.
-func RunIntegrationTest(t *testing.T, test func(ctx context.Context, t *testing.T)) {
+func shouldSkipIntegration(t *testing.T) bool {
+	t.Helper()
+
 	if !strings.HasPrefix(t.Name(), "TestIntegration_") {
 		// We want to enforce a convention so scripts can single out
 		// integration tests.
 		t.Fatalf("Integration tests must have the name format of 'TestIntegration_XXX`")
 	}
 
-	t.Helper()
 	if testing.Short() {
-		t.Skip()
+		t.Skipf("Skipping %s because short tests were requested", t.Name())
 	}
 
-	projID := os.Getenv(EnvDockerRegistry)
-	if projID == "" {
+	if _, ok := os.LookupEnv(EnvDockerRegistry); !ok {
 		t.Skipf("%s is required for integration tests... Skipping...", EnvDockerRegistry)
 	}
 
+	return t.Skipped() || t.Failed()
```

Comment:
nit: You might have the `t.Failed()` for readability, but I don't think it is possible to get to this line and have that be true.